backend/core/src/application-flagged-sets/application-flagged-sets.service.ts

```diff
@@ -115,6 +121,46 @@ export class ApplicationFlaggedSetsService {
     }
   }
 
+  async onApplicationUpdate(
+    newApplication: Application,
+    transactionalEntityManager: EntityManager
+  ) {
+    const transApplicationsRepository = transactionalEntityManager.getRepository(Application)
+    newApplication.markedAsDuplicate = false
+    await transApplicationsRepository.save(newApplication)
+
+    const transAfsRepository = transactionalEntityManager.getRepository(ApplicationFlaggedSet)
+    let afses = await transAfsRepository
+      .createQueryBuilder("afs")
+      .leftJoinAndSelect("afs.applications", "applications")
+      .leftJoinAndSelect("afs.listing", "listing")
+      .where("listing.id = :listingId", { listingId: newApplication.listing.id })
+      .getMany()
+    afses = afses.filter(
+      (afs) => afs.applications.findIndex((app) => app.id === newApplication.id) !== -1
+    )
```

Comment:
Also, do we really need to select all of applications and listing to get what we need for the below operations?

Reply:
I tried doing this but with the following query builder instructions it does not work: 

```
    const qb = transAfsRepository
      .createQueryBuilder("afs")
      .leftJoinAndSelect("afs.applications", "applications")
      .where("afs. = :applicationId", { applicationId: newApplication.id })
```

because the SQL query result contains only an AFS with single (matching id) application. The resulting query looks like this: 

```
"SELECT "afs"."id" AS "afs_id", "afs"."created_at" AS "afs_created_at", "afs"."updated_at" AS "afs_updated_at", "afs"."RULE" AS "afs_rule", "afs"."resolved_time" AS "afs_resolved_time", "afs"."status" AS "afs_status", "afs"."listing_id" AS "afs_listing_id", "afs"."resolving_user_id" AS "afs_resolving_user_id", "applications"."id" AS "applications_id", "applications"."created_at" AS "applications_created_at", "applications"."updated_at" AS "applications_updated_at", "applications"."deleted_at" AS "applications_deleted_at", "applications"."app_url" AS "applications_app_url", "applications"."additional_phone" AS "applications_additional_phone", "applications"."additional_phone_number" AS "applications_additional_phone_number", "applications"."additional_phone_number_type" AS "applications_additional_phone_number_type", "applications"."contact_preferences" AS "applications_contact_preferences", "applications"."household_size" AS "applications_household_size", "applications"."housing_status" AS "applications_housing_status", "applications"."send_mail_to_mailing_address" AS "applications_send_mail_to_mailing_address", "applications"."income_vouchers" AS "applications_income_vouchers", "applications"."income" AS "applications_income", "applications"."income_period" AS "applications_income_period", "applications"."preferences" AS "applications_preferences", "applications"."status" AS "applications_status", "applications"."LANGUAGE" AS "applications_language", "applications"."submission_type" AS "applications_submission_type", "applications"."accepted_terms" AS "applications_accepted_terms", "applications"."submission_date" AS "applications_submission_date", "applications"."marked_as_duplicate" AS "applications_marked_as_duplicate", "applications"."user_id" AS "applications_user_id", "applications"."listing_id" AS "applications_listing_id", "applications"."applicant_id" AS "applications_applicant_id", "applications"."mailing_address_id" AS "applications_mailing_address_id", "applications"."alternate_address_id" AS "applications_alternate_address_id", "applications"."alternate_contact_id" AS "applications_alternate_contact_id", "applications"."accessibility_id" AS "applications_accessibility_id", "applications"."demographics_id" AS "applications_demographics_id" FROM "application_flagged_set" "afs" LEFT JOIN "application_flagged_set_applications_applications" "afs_applications" ON "afs_applications"."application_flagged_set_id"="afs"."id" LEFT JOIN "applications" "applications" ON "applications"."id"="afs_applications"."applications_id" AND ("applications"."deleted_at" IS NULL) WHERE "applications"."id" = $1"
```